app/models/user.rb

```diff
@@ -36,4 +36,8 @@ def self.calculate_win_ratio(won, finished)
     return 0 unless finished > 0
     (won.to_f / finished * 100).round(2)
   end
+
+  def stale?
+    results.order(created_at: :desc).first.created_at < Time.now.utc - 2.weeks
```

Comment:
You're right that IDs (default) aren't always in created at order. One way to ensure order works as intended is to set: `default_scope { order(created_at: :desc) }`